src/main/java/org/junit/runners/ParentRunner.java

```diff
@@ -346,8 +346,12 @@ protected final void runLeaf(Statement statement, Description description,
 
     @Override
     public Description getDescription() {
-        Description description = Description.createSuiteDescription(getName(),
-                getRunnerAnnotations());
+        Description description =
+                Description.createSuiteDescription(
+                        getTestClass().getJavaClass(),
```

Comment:
I see what you mean.

I think it's confusing to add a creational method to `Description` that takes in a class and a name, since the name will only be used if the class parameter is `null`. I would prefer we do this:

    Class<?> clazz = getTestClass().getJavaClass();
    Description description;
    if (clazz == null) {
      description = Description.createSuiteDescription(
              getName(), 
              getRunnerAnnotations());
    } else {
      description = Description.createSuiteDescription(
              getTestClass().getJavaClass(),
              getRunnerAnnotations());

Even then, this would be a change in behavior for subclasses of `ParentRunner` that override `getName()`